crates/network/src/lib.rs

```diff
@@ -33,24 +34,22 @@ pub use any::AnyNetwork;
 
 pub use alloy_eips::eip2718;
 
-/// A list of transactions, either hydrated or hashes.
-#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
-#[serde(untagged)]
-pub enum TransactionList<T> {
-    /// Hashes only.
-    Hashes(Vec<B256>),
-    /// Hydrated tx objects.
-    Hydrated(Vec<T>),
-    /// Special case for uncle response
-    Uncled,
+/// A block response.
+pub trait BlockResponse<N: Network> {
+    /// Header of the block.
+    fn header(&self) -> &N::HeaderResponse;
+    /// Block transactions.
+    fn transactions(&self) -> &TransactionList<N::TransactionResponse>;
 }
 
-/// A block response
-#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
-pub struct BlockResponse<N: Network> {
-    #[serde(flatten)]
-    header: N::HeaderResponse,
-    transactions: TransactionList<N::TransactionResponse>,
+/// A header response.
+pub trait HeaderResponse {
+    /// Base fee per unit of gas (if past London)
+    fn base_fee_per_gas(&self) -> &Option<U256>;
+    /// Returns the blob fee for the next block according to the EIP-4844 spec.
+    ///
+    /// Returns `None` if `excess_blob_gas` is None.
+    fn next_block_blob_fee(&self) -> Option<u128>;
```

Comment:
for header we need a few more things, but all of these should be moved to a header trait I believe